modules/core/src/lib/base-attribute.js

```diff
@@ -156,6 +160,31 @@ export default class BaseAttribute {
     }
   }
 
+  // https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/vertexAttribPointer
+  _normalizeConstant(value) {
+    switch (this.type) {
+      case GL.BYTE:
+        // normalize [-128, 127] to [-1, 1]
+        return value.map(x => ((x + 128) / 255) * 2 - 1);
```

Comment:
Can't `value` be a `Uint8Array`? If so, these divisions won't work (since Uint8 cannot store floats).

Based on how I would have expected to do this, this "normalization" should not be needed, since if we set the attribute's `normalize` flag, this calculation should be performed by the GPU.

Perhaps we should add a `Uint8Array` to the test cases (as far as I could tell, they currently only have standard JS arrays).

Reply:
Yeah it's possible, I can explicitly case it to Float32Array.

Only the buffered attributes are normalized; generic attributes are not. I've tested in Chrome, Firefox and Safari.

Reply:
As far as I can tell `normalized` is only applied when calling `gl. vertexAttribPointer`. When the attribute is disabled for a generic value the flag is no longer respected.